Framework/HistogramData/test/HistogramItemTest.h

```diff
@@ -0,0 +1,160 @@
+#ifndef MANTID_HISTOGRAMDATA_HISTOGRAMITEMTEST_H_
+#define MANTID_HISTOGRAMDATA_HISTOGRAMITEMTEST_H_
+
+#include <cxxtest/TestSuite.h>
+
+#include "MantidHistogramData/Histogram.h"
+#include "MantidHistogramData/HistogramItem.h"
+
+using Mantid::HistogramData::Histogram;
+using Mantid::HistogramData::HistogramItem;
+using Mantid::HistogramData::Points;
+using Mantid::HistogramData::Counts;
+using Mantid::HistogramData::Frequencies;
+using Mantid::HistogramData::BinEdges;
+
+class HistogramItemTest : public CxxTest::TestSuite {
+public:
+  // This pair of boilerplate methods prevent the suite being created statically
+  // This means the constructor isn't called when running other tests
+  static HistogramItemTest *createSuite() { return new HistogramItemTest(); }
+  static void destroySuite(HistogramItemTest *suite) { delete suite; }
+
+  static constexpr double tolerance = 1e-6;
+
+  void test_construction() {
+    Histogram hist(Histogram::XMode::BinEdges, Histogram::YMode::Counts);
+    TS_ASSERT_THROWS_NOTHING(HistogramItem item(hist, 0));
+  }
+
+  void test_get_counts_from_histogram_with_counts() {
+    Histogram hist(Points{0.1, 0.2, 0.4}, Counts{1, 2, 4});
+    HistogramItem item(hist, 1);
+    TS_ASSERT_EQUALS(item.counts(), 2.0)
+  }
+
+  void test_get_counts_from_histogram_with_frequencies() {
+    Histogram hist(Points{0.1, 0.2, 0.4}, Frequencies{1, 2, 4});
+    HistogramItem item(hist, 1);
+    TS_ASSERT_DELTA(item.counts(), 0.3, tolerance)
+  }
+
+  void test_get_countVariance_from_histogram_with_counts() {
+    Histogram hist(Points{0.1, 0.2, 0.4}, Counts{1, 2, 4});
+    HistogramItem item(hist, 1);
+    TS_ASSERT_DELTA(item.countVariance(), 2.0, tolerance)
+  }
+
+  void test_get_countVariance_from_histogram_with_frequencies() {
+    Histogram hist(Points{0.1, 0.2, 0.4}, Frequencies{1, 2, 4});
+    HistogramItem item(hist, 1);
+    TS_ASSERT_DELTA(item.countVariance(), 0.045, 1e-12)
```

Comment:
I think I would write most (or all?) of these test simply comparing to elements of the corresponding method on `Histogram`, for example:
```cpp
TS_ASSERT_DELTA(item.countVariance(), hist.countVariance()[index]);
```

Edit:
Maybe doing all the testing on the iterator is more useful/simple, testing the whole range? Consider changing `HistogramItem` such that it can be constructed only by the iterator (no one else needs to?).